frontend/src/shared/components/register-page/register-page.component.ts

```diff
@@ -0,0 +1,73 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { faGithub, faGoogle } from '@fortawesome/free-brands-svg-icons';
+
+@Component({
+  selector: 'app-register-page',
+  templateUrl: './register-page.component.html',
+  styleUrls: ['./register-page.component.sass']
+})
+export class RegisterPageComponent implements OnInit {
+
+  public name = '';
+  public email = '';
+  public password = '';
+  public passwordRepeat = '';
+  public hidePass = true;
+  public hidePassRepeat = true;
+  public registerForm: FormGroup =  new FormGroup({});
+  public nameControl: FormControl;
+  public emailControl: FormControl;
+  public passwordControl: FormControl;
+  public passwordRepeatControl: FormControl;
+  faGithub = faGithub;
+  faGoogle = faGoogle;
+
+  constructor() { 
+    this.nameControl = new FormControl(this.name, [
+      Validators.required,
+    ]);
+    this.emailControl = new FormControl(this.email, [
+      Validators.email,
+      Validators.required,
+      Validators.minLength(8)
```

Comment:
we should probably add something like validation.constants file where these values will be stored

Reply:
Something like interface with this data to use it in other components? As I understand after registration email with confirmation message should be received and then with login we will get token

Reply:
I meant the amount of symbols (min/max length) :)